core/src/main/java/org/arquillian/cube/impl/client/CubeExtension.java

```diff
@@ -31,17 +32,22 @@ public void register(ExtensionBuilder builder) {
 
         // Arquillian Container integration
         // Only register if container-test-spi is on classpath
-        if(Validate.classExists("org.jboss.arquillian.container.test.spi.client.deployment.AuxiliaryArchiveAppender")) {
+        if (Validate.classExists("org.jboss.arquillian.container.test.spi.client.deployment.AuxiliaryArchiveAppender")) {
             builder.observer(ProtocolMetadataUpdater.class)
                    .observer(CubeContainerLifecycleController.class)
                    .observer(ContainerConfigurationController.class)
                    .observer(CubeRemoteCommandObserver.class);
             builder.service(AuxiliaryArchiveAppender.class, CubeAuxiliaryArchiveAppender.class);
         }
         // Only register if container-test-impl is on classpath
-        if(Validate.classExists("org.jboss.arquillian.container.test.impl.enricher.resource.OperatesOnDeploymentAwareProvider")) {
+        if (Validate.classExists("org.jboss.arquillian.container.test.impl.enricher.resource.OperatesOnDeploymentAwareProvider")) {
             builder.service(ResourceProvider.class, CubeIDResourceProvider.class);
         }
+
+        // Only if recorder-reporter is in classpath we should provide reporting capabilities.
+        if (Validate.classExists("org.arquillian.recorder.reporter.ReporterExtension")) {
+            builder.observer(TakeCubeInformation.class);
+        }
```

Comment:
As a general thought - I'm not sure if this way of validation is the best one. If we change package names or shuffle classes around, or even change their names this will be all broken. We should think about some other way of detection in the nearest future.